pkg/provider/core.go

```diff
@@ -430,6 +441,28 @@ func FindInstanceByFolderAndName(ctx context.Context, sdk *ycsdk.SDK, folderID s
 	return result.Instances[0], nil
 }
 
+func findUnusedAddressFromListByFolder(ctx context.Context, sdk *ycsdk.SDK, folderID string, addresses []string) (string, error) {
+	result, err := sdk.VPC().Address().List(ctx, &vpc.ListAddressesRequest{
+		FolderId: folderID,
+	})
+	if err != nil {
+		return "", err
+	}
+
+	for _, item := range result.Addresses {
+		if !item.GetUsed() {
+			external := item.GetExternalIpv4Address().GetAddress()
```

Comment:
we can add `if external==""` then continue , to avoid the extra looping

Reply:
Please check again